rust/src/lockfile.rs

```diff
@@ -118,20 +118,25 @@ struct LockfileConfigMetadata {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
```

Comment:
This is likely not a good idea as it ossifies the data-structure, making it impossible to introduce new fields without a strict software<->lockfile version interlock (if we need to, in the future).

Perhaps you were looking for something like https://docs.rs/serde_ignored instead, to detect and warn about unknown fields?

Reply:
Ah yes, that looks like a nice crate.  But I don't follow this argument:

> making it impossible to introduce new fields without a strict software<->lockfile version interlock 

How so?  The semantics desired here are error (or warning) if new key is present with old rpm-ostree.  Newer rpm-ostree will accept the key.

Reply:
IOW, this isn't `deny_missing_fields` - since the new key is an `Option<T>` old lockfiles will work fine with newer rpm-ostree.

Reply:
`deny_unknown_fields` matches what we do for the treefile today. The reason I want it (and I suspect also why the treefile uses it as well) is that triggering it means you've either mistyped something or you're trying to use features of rpm-ostree which are not supported by the version you're running. Either way, the end result is that rpm-ostree likely will not do what you expect it to (though the `metadata` key is a counterexample of that because rpm-ostree just ignores it, but that's basically the first of its kind :) ).

Reply:
I'm fine with the arguments and merging, just voiced my concern with `error if new key is present with old rpm-ostree` eating possible wiggle room for the future.